src/content/en/fundamentals/getting-started/primers/media-source-extensions.md

```diff
@@ -0,0 +1,323 @@
+    
+    function sourceOpen(e) {
+      var mime = 'video/webm; codecs="opus, vp9"';
+      var mediaSource = this;
+      var sourceBuffer = mediaSource.addSourceBuffer(mime);
+      var video = 'droid.webm';
+      fetch(video)
+        .then(function(response) {
+          return response.arrayBuffer();
+        })
+        .then(function(arrayBuffer) {
+          sourceBuffer.onupdateend = function(e) {
+            if (sourceBuffer.updating && mediaSource.readyState === 'open') {
+              mediaSource.endOfStream();
+            }
+          }
+          sourceBuffer.appendBuffer(arrayBuffer);
+        });
+    }
+
+  
+If you can sort things out from the explanations so far, feel free to stop
+reading now. If you want a more detailed explanation, then please keep reading.
+I'm going to walk through this chain by building a simple MSE example. Each of
+the build steps will add code to the previous step. 
+
+### A note about clarity
+
+Will this article tell you everything you need to know about playing media on a
+web page? No, it's only intended to help you understand more complicated code
+you might find elsewhere. For the sake of clarity, this document simplifies and
+excludes many things. We think we can get away with this because we also
+recommend using a library such as [Google's Shaka Player](https://shaka-player-demo.appspot.com/demo/).
+I will note throughout where I'm deliberately simplifying.
+
+### A few things not covered
+
+Here, in no particular order, are a few things I won't cover.
+
++  Playback controls. We get those for free by virtue of using the HTML5
+   `<audio>` and `<video>` elements.
++  Error handling for the sake of simplicyt and clarity.
+
+## Attach a MediaSource instance to a media element
+
+As with many things in web development these days, you start with feature
+detection. Next, get a media element, either an `<audio>` or `<video>` element.
+Finally create an instance of `MediaSource`. It gets turned into a URL and passed
+to the media element's source attribute.
+
+    var vidElement = document.querySelector('video');
+
+    if (window.MediaSource) {
+      var mediaSource = new MediaSource();
+      vidElement.src = URL.createObjectURL(mediaSource);
+      // Is the MediaSource instance ready?
+    } else {
+      console.log("The Media Source Extensions API is not supported.")
+    }
+
+Note: Each incomplete code example contains a comment that gives you a hint of
+what I'll add in the next step. In the example above, this comment says, 'Is the
+MediaSource instance ready?', which matches the title of the next section.
+
+That a `MediaSource` object can be passed to a `src` attribute might seem a bit
+odd. Aren't `src` attributes usually strings? A media element's `src`
+[attribute can also be a blob](https://www.w3.org/TR/FileAPI/#url).
+If you inspect a page with embedded media and examine its media element, you'll
+see what I mean.
+
+![A source attribute as a blob](imgs/media-url.png)
```

Comment:
Shift this up above "That a media source..." , change to HTML and add `class="attempt-right"` so that it flows within the doc body. You'll likely need a `<div class="clear-fix"></div>`  before the new heading.